cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -582,6 +583,29 @@ func (m *McmManager) GetMachineDeploymentNodes(machinedeployment *MachineDeploym
 	return nodes, nil
 }
 
+// validateNodeTemplate function validates the NodeTemplate object of the MachineClass
+func validateNodeTemplate(nodeTemplateAttributes *v1alpha1.NodeTemplate) error {
+	var allErrs []error
+
+	capacityAttributes := []v1.ResourceName{"cpu", "gpu", "memory"}
+
+	for _, attribute := range capacityAttributes {
+		if _, ok := nodeTemplateAttributes.Capacity[attribute]; !ok {
+			errMessage := fmt.Errorf("CPU, GPU and memory fields are mandatory")
+			klog.Warning(errMessage)
+			allErrs = append(allErrs, errMessage)
+		}
+	}
+
+	if nodeTemplateAttributes.Region == "" || nodeTemplateAttributes.InstanceType == "" || nodeTemplateAttributes.Zone == "" {
+		errMessage := fmt.Errorf("InstanceType, Region and Zone attributes are mandatory")
+		klog.Warning(errMessage)
+		allErrs = append(allErrs, errMessage)
```

Comment:
same error and warning log repetition

Reply:
This `if` statement is independent of the loop above. So, the warning should only print the one in the context and should get appended to `allErrs`.

Reply:
Oh yes its an `or` ,my bad.